core/dbt/task/init.py

```diff
@@ -274,9 +304,18 @@ def run(self):
             in_project = False
 
         if in_project:
+            # If --profile was specified, it means use an existing profile, which is not
+            # applicable to this case
+            if self.args.profile:
+                raise DbtRuntimeError(
+                    msg="Can not init existing project with specified profile, edit dbt_project.yml instead"
+                )
+
             # When dbt init is run inside an existing project,
             # just setup the user's profile.
-            profile_name = self.get_profile_name_from_current_project()
+            if not self.args.skip_profile_setup:
+                profile_name = self.get_profile_name_from_current_project()
+                self.setup_profile(profile_name)
```

Comment:
@ezraerb Can you help me understand why we'd want or need `self.get_profile_name_from_current_project()` to be guarded by `if not self.args.skip_profile_setup:`?

Could we safely do the following outside of `if` statement?
```python
            profile_name = self.get_profile_name_from_current_project()
```

If so, then we could avoid repeating `self.setup_profile(profile_name)` twice (once for each case) and instead we can just do it once at the very end (IFF profile setup is being skipped).

Reply:
There are four cases that need to be considered:
1. Initializing inside an existing project, no profile specified. The profile is created (the code being commented on)
2. Initializing inside an existing project, profile specified. This raises an error.
3. Initializing outside an existing project, no profile specified. Create the project and then create the profile.
4. Initializing outside an existing project, profile specified. Create the project only.

The difference between cases 3 and 4 prohibits having a catch-all profile creation at the end of the method. Either signal variables are needed or the calls need to be duplicated in the various paths. Previous viewers expressed a desire for the latter as they believe it is easier to follow.